### PR TITLE
Add a placeholder when folder is empty in folder browser

### PR DESCRIPTION
**Pull request by zheyunwu**
For card: https://app.clickup.com/t/1h256vq

### CODE REVIEW

**Comment by giardiv** [comment]
Task linked: [CU-1h256vq add a placeholder when folder is empty in Folder browser](https://app.clickup.com/t/1h256vq)

**Review by snqb — APPROVED** [review]

File: src/components/file-system-explorer/empty-folder-info/empty-folder-info.styles.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  minHeight: '62px',
+  width: '100%',
+  minWidth: '230px',
+  p: '20px',
+  boxSizing: 'border-box',
+  '>h2': {
+    fontFamily: 'subtitle',
+    fontWeight: 'subtitle',
+    fontSize: 'subtitle',
+    color: '#A0A0A0',
```
Comment (snqb) [inline src/components/file-system-explorer/empty-folder-info/empty-folder-info.styles.ts]: this is a `gray` color defined in the theme

File: src/components/file-system-explorer/column/index.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@ export interface FileExplorerColumnProps
   selectPathListValue?: any;
   fileListValue?: any[];
   validExtensions?: string[];
+  setIsEmptyFolder?: any;
```
Comment (snqb) [inline src/components/file-system-explorer/column/index.tsx]: it isn't any, it's a `(value: boolean) => void`

File: src/components/file-system-explorer/file/index.tsx [hunk 3]
```diff
@@ -35,12 +36,14 @@ const FileExplorer: FC<QuartzFileExplorerProps> = ({
   selected,
   isFileSelected,
   isValidExtension,
+  setIsEmptyFolder,
```
Comment (snqb) [inline src/components/file-system-explorer/file/index.tsx:39]: this is a 3rd prop with a setter
have you considered Context here? We shouldn't be drilling props all the time imo

File: src/components/file-system-explorer/index.tsx [hunk 4]
```diff
@@ -70,13 +71,22 @@ const FileSystemExplorer: FC<FileSystemExplorerProps> = ({
   const contentRef = useRef<HTMLDivElement>();
 
   const [columns, setColumns] = useState(data);
+  const [isEmptyFolder, setIsEmptyFolder] = useState<boolean>(false);
 
   useEffect(() => {
-    setColumns(data);
+    if (Array.isArray(data)) {
+      if (Array.isArray(data[data.length - 1]) && data[data.length - 1].length === 0) {
```
Comment (snqb) [inline src/components/file-system-explorer/index.tsx]: can we extract `data[data.length - 1]` in a variable?

File: src/components/file-system-explorer/index.tsx [hunk 5]
```diff
@@ -70,13 +71,22 @@ const FileSystemExplorer: FC<FileSystemExplorerProps> = ({
...
+      } else {
+        setIsEmptyFolder(false);
+        setColumns(data);
+      }
+    }
   }, [data]);
 
   const [activeFile, setActiveFile] = useState<ActiveFile>();
-  //@ts-ignore
+  // @ts-ignore
```
Comment (snqb) [inline src/components/file-system-explorer/index.tsx]: what do we need this `@ts-ignore` here for?

**Review by snqb — CHANGES_REQUESTED** [review]
The `any` is not good, it is easily avoidable
also, there are quite a bit of smaller comments

File: src/components/file-system-explorer/index.tsx (hunk 5 above)
Comment (zheyunwu) [reply src/components/file-system-explorer/index.tsx]: Not sure, it was there before, here just added a space to escape the eslint error msg. 
Without it there seems no differences, should I remove it?

File: src/components/file-system-explorer/file/index.tsx (hunk 3 above)
Comment (zheyunwu) [reply src/components/file-system-explorer/file/index.tsx:39]: Here I followed other props like "setActiveFile" and "setActiveFolder" that behaves the same way, I can look into how to optimise them together without breaking something later.

**Review by giardiv — APPROVED** [review]
Works fine

**Review by snqb — APPROVED** [review]

**Pull request merged**